sherpa/utils/tests/test_utils.py

```diff
@@ -263,16 +264,16 @@ def test_neville2d(self):
                 val = utils.neville2d(xx, yy, x, y, fval)
                 self.assertTrue(utils.Knuth_close(answer, val, tol))
 
-    def test_parallel_map(self):
-        ncpus = multiprocessing.cpu_count()
 
+@pytest.mark.parametrize("num_tasks", [1, 3, 100])
```

Comment:
I htink it would be good to iterate over both `num_tasks` (or whatever it gets called, see next comment) and `numtasks`, so that we have covered at least:

- (1, 1)
- (n > 1, 1)
- (1, n > 1)
- (n, m) where n > m
- (n, m) where n < m
- (n, n) for n > 1

just to make sure we've covered the corner cases of more tasks than processes, more processes than tasks, the same number of tasks and processes, ...